src/guest_tracking.rs

```diff
@@ -91,8 +91,15 @@ impl<T: GuestStagePagingMode> Clone for GuestVm<T> {
 }
 
 impl<T: GuestStagePagingMode> GuestVm<T> {
+    /// Return required pages necessary to create a GuestVM.
+    pub const fn required_pages() -> u64 {
+        PageArc::<RwLock<GuestVmInner<T>>>::required_pages()
+    }
+
     /// Creates a new initializing `GuestVm` from `vm`, using `page` as storage.
     pub fn new(vm: Vm<T>, page: Page<InternalClean>) -> Self {
+        // assert it fits in a single page for now.
+        assert!(Self::required_pages() == 1);
```

Comment:
This was a line that was added to let the preparation commit compile in case of bisect, but it is immediately overwritten. Didn't know `const_assert` though. Thanks!